tests/scalers/dynatrace/dynatrace_test.go

```diff
@@ -0,0 +1,333 @@
+kind: Deployment
+metadata:
+  name: {{.DeploymentName}}
+  namespace: {{.TestNamespace}}
+  labels:
+    app: {{.DeploymentName}}
+spec:
+  replicas: 0
+  selector:
+    matchLabels:
+      app: {{.DeploymentName}}
+  template:
+    metadata:
+      labels:
+        app: {{.DeploymentName}}
+    spec:
+      containers:
+      - name: prom-test-app
+        image: tbickford/simple-web-app-prometheus:a13ade9
+        imagePullPolicy: IfNotPresent
+`
+
+	serviceTemplate = `apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    name: {{.ServiceName}}
+  annotations:
+    prometheus.io/scrape: "true"
+  name: {{.ServiceName}}
+  namespace: {{.TestNamespace}}
+spec:
+  ports:
+  - name: http
+    port: 80
+    protocol: TCP
+    targetPort: 8080
+  selector:
+    app: {{.MonitoredDeploymentName}}
+  `
+
+	scaledObjectTemplate = `apiVersion: keda.sh/v1alpha1
+kind: ScaledObject
+metadata:
+  name: {{.ScaledObjectName}}
+  namespace: {{.TestNamespace}}
+  labels:
+    app: {{.DeploymentName}}
+spec:
+  scaleTargetRef:
+    name: {{.DeploymentName}}
+  minReplicaCount: {{.MinReplicaCount}}
+  maxReplicaCount: {{.MaxReplicaCount}}
+  pollingInterval: 1
+  cooldownPeriod:  1
+  advanced:
+    horizontalPodAutoscalerConfig:
+      behavior:
+        scaleDown:
+          stabilizationWindowSeconds: 10
+  triggers:
+    - type: dynatrace
+      metadata:
+        host: {{.DynatraceHost}}
+        threshold: "2"
+        activationThreshold: "3"
+	metricSelector: "builtin:service.requestCount.total:splitBy():fold"
+	from: now-2m
+      authenticationRef:
+        name: {{.TriggerAuthName}}
+`
+
+	lightLoadTemplate = `apiVersion: v1
+kind: Pod
+metadata:
+  name: fake-light-traffic
+  namespace: {{.TestNamespace}}
+spec:
+  containers:
+  - image: busybox
+    name: test
+    command: ["/bin/sh"]
+    args: ["-c", "while true; do wget -O /dev/null -o /dev/null http://{{.ServiceName}}/; sleep 0.5; done"]`
+
+	heavyLoadTemplate = `apiVersion: v1
+kind: Pod
+metadata:
+  name: fake-heavy-traffic
+  namespace: {{.TestNamespace}}
+spec:
+  containers:
+  - image: busybox
+    name: test
+    command: ["/bin/sh"]
+    args: ["-c", "while true; do wget -O /dev/null -o /dev/null http://{{.ServiceName}}/; sleep 0.1; done"]`
+)
+
+func TestDynatraceScaler(t *testing.T) {
+	// setup
+	t.Log("--- setting up ---")
+	require.NotEmpty(t, dynatraceToken, "DYNATRACE_METRICS_TOKEN env variable is required for dynatrace tests")
+	require.NotEmpty(t, dynatraceHost, "DYNATRACE_HOST env variable is required for dynatrace tests")
+
+	kc := GetKubernetesClient(t)
+	data, templates := getTemplateData()
+	t.Cleanup(func() {
+		DeleteKubernetesResources(t, testNamespace, data, templates)
+	})
+
+	// Create kubernetes resources
+	CreateKubernetesResources(t, kc, testNamespace, data, templates)
+
+	installDynatrace(t)
+
+	data, templates = getDynatraceTemplateData()
+	// Create Dynatrace-specific kubernetes resources
+	CreateKubernetesResources(t, kc, testNamespace, data, templates)
```

Comment:
CreateKubernetesResources recreates the namespace too (deleting the previous resources)
```suggestion
	KubectlApplyMultipleWithTemplate(t, data, templates)
```